distributed/client.py

```diff
@@ -578,14 +597,16 @@ def start(self, asynchronous=None, **kwargs):
                 self._should_close_loop = True
             while not self.loop._running:
                 sleep(0.001)
+
         pc = PeriodicCallback(lambda: None, 1000, io_loop=self.loop)
         self.loop.add_callback(pc.start)
         _set_global_client(self)
-        if asynchronous:
+        self.status = 'connecting'
+
+        if self.asynchronous:
```

Comment:
> The client in both cases is the same. 

Does it have to? :-) I don't think it would be terribly ruinous to restrict that asynchronous clients cannot be re-used in a synchronous context, and vice-versa.